esmvalcore/preprocessor/_bias.py

```diff
@@ -0,0 +1,114 @@
+"""Preprocessor functions to calculate biases from data."""
+import logging
+
+import dask.array as da
+import iris.cube
+
+from ._io import concatenate
+
+logger = logging.getLogger(__name__)
+
+
+def bias(products, bias_type='absolute', denominator_mask_threshold=1e-3,
+         keep_reference_dataset=False):
+    """Calculate biases.
+
+    Notes
+    -----
+    This preprocessor requires a reference dataset. For this, exactly one input
+    dataset needs to have the facet ``reference_for_bias: true`` defined in the
+    recipe. In addition, all input datasets need to have identical dimensional
+    coordinates. This can for example be ensured with the preprocessors
+    :func:`esmvalcore.preprocessor.regrid` and/or
+    :func:`esmvalcore.preprocessor.regrid_time`.
+
+    Parameters
+    ----------
+    products: set of esmvalcore.preprocessor.PreprocessorFile
+        Input datasets. Exactly one datasets needs the facet
+        ``reference_for_bias: true``.
+    bias_type: str, optional (default: 'absolute')
+        Bias type that is calculated. Must be one of ``'absolute'`` (dataset -
+        ref) or ``'relative'`` ((dataset - ref) / ref).
+    denominator_mask_threshold: float, optional (default: 1e-3)
+        Threshold to mask values close to zero in the denominator (i.e., the
+        reference dataset) during the calculation of relative biases. All
+        values in the reference dataset with absolute value less than the given
+        threshold are masked out. This setting is ignored when ``bias_type`` is
+        set to ``'absolute'``.
```

Comment:
are there any variables that have have both negative and positive values and their range crosses 0? Because those, if there are any, should be treated with a lot of caution when biases are computed. Also, variables with tiny values - same stuff, maybe you can add a note on the fact that the threshold should be these many orders smaller than varaibles values that hover around 0

Reply:
Very good point! I will put a note in the doc. An example are many carbon flux variables, which can be positive AND negative in the order of `1e-7`; in this case the default values will produce a fully masked globe.